### PR TITLE
Switchable switchers automation

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherCompany.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.hedvig.contract.shared.switchers
+
+enum class SwitchableSwitcherCompany(val realName: String, val email: String) {
+  FOLKSAM("Folksam", "annullationer-sak@folksam.se"),
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherCompany.kt]: Config maybe? i am not sure though how often these emails will change?

File: src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherEmailEventListener.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.axonframework.config.ProcessingGroup
+import org.axonframework.eventhandling.EventHandler
+import org.springframework.stereotype.Component
+import java.util.UUID
+
+@Component
+@ProcessingGroup("SwitchableSwitcherEmails")
+class SwitchableSwitcherEmailEventListener(
+  private val switcherEmailService: SwitcherEmailService,
+  private val userRepository: UserRepository
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherEmailEventListener.kt]: This is not used anymore, so you will not find all the new members. Its better if you call member-service

File: src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmail.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import javax.persistence.Id
+import javax.persistence.OneToOne
+
+@Entity
+class SwitcherEmail {
+  @Id
+  lateinit var id: UUID
+
+  @OneToOne
+  lateinit var user: UserEntity
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmail.kt]: Check my comment below

File: src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherCompany.kt (hunk 1 above)
Comment (palmenhq) [reply src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherCompany.kt]: I think next-to-never. And don't you think it'd be better to have it versioned which emails we send to? :)

File: src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherEmailEventListener.kt (hunk 2 above)
Comment (palmenhq) [reply src/main/java/com/hedvig/contract/shared/switchers/SwitchableSwitcherEmailEventListener.kt]: ojdå! Thanks. Lets mark it as deprecated too then :)

File: src/main/java/com/hedvig/contract/shared/switchers/mail/SwitcherCancellationMailerImpl.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  private val logger: Logger = getLogger(SwitcherCancellationMailerImpl::class.java)
+
+  override fun sendEmail(switcherCompany: SwitchableSwitcherCompany, memberId: String, ssn: String, insuranceMandatePdfContent: ByteArray): String {
+    logger.info("Composing cancellation email to $switcherCompany for member $memberId...")
+
+    val helper = MimeMessageHelper(mailSender.createMimeMessage(), true)
+
+    helper.setFrom("Hedvig <$fromAddress>")
+    helper.setSubject("Huvudförfallodag och uppsägning")
+    val mailHtml = getMailHtml(ssn)
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/shared/switchers/mail/SwitcherCancellationMailerImpl.kt]:
```suggestion
    val mailHtml = renderMailHtml(ssn)
```
maybe?

File: src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmailRepository.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.UUID
+
+interface SwitcherEmailRepository : CrudRepository<SwitcherEmail, UUID> {
+  @Query("""
+    SELECT DISTINCT se
+    FROM SwitcherEmail se
+    INNER JOIN ContractEntity ce ON ce.memberId = se.user.id
+    LEFT JOIN apartment_contract ac ON ac.holderMemberId = se.user.id
+    LEFT JOIN house_contract hc ON hc.holderMemberId = se.user.id
+    WHERE hc.agreements.size = 0
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmailRepository.kt]: Why de we want the agreements.size to be 0

File: src/main/java/com/hedvig/contract/shared/switchers/mail/RealMailSendStrategy.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.hedvig.contract.shared.switchers.mail
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.mail.javamail.JavaMailSender
+import org.springframework.mail.javamail.MimeMessageHelper
+import org.springframework.stereotype.Component
+
+@ConditionalOnProperty(name = [MAIL_SENDING_STRATEGY_PROP], havingValue = "SEND_REAL")
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/shared/switchers/mail/RealMailSendStrategy.kt]: Super nice! 👍

File: src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmailRepository.kt (hunk 5 above)
Comment (palmenhq) [reply src/main/java/com/hedvig/contract/shared/switchers/persistence/SwitcherEmailRepository.kt]: If they have an agreement (that isn't pending) they should no longer be included among the ones to receive emails :)

**Pull request merged**